### PR TITLE
allow refresh token removal

### PR DESCRIPTION
**Pull request by duytiennguyen-okta**
Allow refresh_token removal by switching check from refresh_token_leeway to actual check in grant types

### CODE REVIEW

File: okta/resource_okta_app_oauth.go [hunk 1]
```diff
@@ -770,21 +770,31 @@ func buildAppOAuth(d *schema.ResourceData) *sdk.OpenIdConnectApplication {
...
 	if leeway, ok := d.GetOk("refresh_token_leeway"); ok {
 		refresh.LeewayPtr = int64Ptr(leeway.(int))
-		hasRefresh = true
 	}
 
 	if hasRefresh {
 		app.Settings.OauthClient.RefreshToken = refresh
 	}
 
+	// TODU
```
Comment (duytiennguyen-okta) [inline okta/resource_okta_app_oauth.go]: Should we add an error or a warning here?

File: okta/resource_okta_app_oauth.go [hunk 2]
```diff
@@ -770,21 +770,31 @@ func buildAppOAuth(d *schema.ResourceData) *sdk.OpenIdConnectApplication {
...
+	// if !hasRefresh && refresh != nil {
+	// 	return nil, errors.New("does not have refresh grant type but refresh_token_rotation and refresh_token_leeway exist in payload")
+	// }
+
+	// TODU
+	// should we unset here refresh_token_rotation, refresh_token_leeway?
```
Comment (duytiennguyen-okta) [inline okta/resource_okta_app_oauth.go]: Should we unset refresh_token_rotation and refresh_token_leeway from the resource if it does not include the refresh?

File: okta/resource_okta_app_oauth_test.go [hunk 3]
```diff
@@ -87,10 +87,11 @@ func TestAccResourceOktaAppOauth_refreshToken(t *testing.T) {
 	//       Skipping for now assuming that the okta account doesn't have this feature enabled.
 	//       If this feature is enabled or Okta releases this to all this test should be enabled.
 	//       SEE https://help.okta.com/en/prod/Content/Topics/Apps/apps-fbm-enable.htm
-	t.Skip("This is an 'Early Access Feature' and needs to be enabled by Okta, skipping this test as it fails when this feature is not available")
+	// t.Skip("This is an 'Early Access Feature' and needs to be enabled by Okta, skipping this test as it fails when this feature is not available")
```
Comment (monde) [inline okta/resource_okta_app_oauth_test.go]: I'd delete this line

File: okta/resource_okta_app_oauth.go [hunk 4]
```diff
@@ -770,21 +770,31 @@ func buildAppOAuth(d *schema.ResourceData) *sdk.OpenIdConnectApplication {
 	}
 
 	refresh := &sdk.OpenIdConnectApplicationSettingsRefreshToken{}
-	hasRefresh := false
+	var hasRefresh bool
+	for _, grant_type := range grantTypes {
+		if grant_type == refreshToken {
+			hasRefresh = true
```
Comment (monde) [inline okta/resource_okta_app_oauth.go:776]: there could be a break after this `hasRefresh = true`

File: okta/resource_okta_app_oauth.go (hunk 1 above)
Comment (monde) [reply okta/resource_okta_app_oauth.go]: if it should have have a refresh value, but does not, that is an error is it not?

File: okta/resource_okta_app_oauth_test.go (hunk 3 above)
Comment (duytiennguyen-okta) [reply okta/resource_okta_app_oauth_test.go]: So delete the skip or not?

**Review by monde — APPROVED** [review]

**Pull request merged**